minimal.js

```diff
@@ -83,36 +82,50 @@ module.exports = function (dirname, keys, opts) {
 
   var append = db.rawAppend = db.append
   db.post = Obv()
-  var queue = AsyncWrite(function (_, cb) {
-    var batch = state.queue
-    state.queue = []
-    append(batch, function (err, v) {
+
+  let writing = false
+
+  const write = () => {
+    writing = true
+    // Very defensive: Is this necessary? I don't know whether it's possible
+    // for another function to `state.queue.push()` between these two lines.
+    const batch = state.queue.slice()
+    state.queue = state.queue.slice(batch.length)
+
+    append(batch, function (err) {
+      // Previously this error wasn't being caught anywhere. :(
+      // New behavior: if a write fails, crash loudly and throw an error.
+      if (err) throw err
+
+      // If we have new messages in the queue, write them!
+      // Otherwise, run all callbacks added via `flush.add()`
+      if (state.queue.length) {
+        write()
+      } else {
+        writing = false
+        flush.runAll()
+      }
+
+      // Update the observable
       batch.forEach(function (data) {
         db.post.set(u.originalData(data))
       })
-      cb(err, v)
     })
-  }, function reduce (_, msg) {
-    return V.append(state, hmacKey, msg)
-  }, function (_state) {
-    return state.queue.length > 1000
-  }, function isEmpty (_state) {
-    return !state.queue.length
-  }, 100)
-
-  queue.onDrain = function onDrain () {
-    if (state.queue.length === 0) {
-      flush.runAll()
+  }
+
+  const queue = (message, cb) => {
+    try {
+      V.append(state, hmacKey, message)
```

Comment:
I think I'd like a note which tells readers that `V.append` mutates `state.queue`. I know it's not your code, but it would make maintenance + readability easier

Reply:
<3